Source/WebCore/Modules/url-pattern/URLPattern.cpp

```diff
@@ -26,29 +26,239 @@
+        }
+
+        if (init.protocol.isNull()
+            && init.hostname.isNull()
+            && init.port.isNull()) {
+            auto port = baseURL.port();
+            result.port = port ? String::number(*port) : ""_s;
+        }
+
+        if (init.protocol.isNull()
+            && init.hostname.isNull()
+            && init.port.isNull()
+            && init.pathname.isNull()) {
+            String serializedBaseURL = baseURL.path().toString();
+            result.pathname = processBaseURLString(serializedBaseURL, type);
+        }
+
+        if (init.protocol.isNull()
+            && init.hostname.isNull()
+            && init.port.isNull()
+            && init.pathname.isNull()
+            && init.search.isNull()) {
+            String urlQuery = baseURL.hasQuery() ? baseURL.query().toString() : ""_s;
+            result.search = processBaseURLString(urlQuery, type);
+        }
+
+        if (init.protocol.isNull()
+            && init.hostname.isNull()
+            && init.port.isNull()
+            && init.pathname.isNull()
+            && init.search.isNull()
+            && init.hash.isNull()) {
+            String urlFragment = baseURL.hasFragmentIdentifier() ? baseURL.fragmentIdentifier().toString() : ""_s;
+            result.hash = processBaseURLString(urlFragment, type);
+        }
+    }
+
+    if (!init.protocol.isNull()) {
+        ExceptionOr<String> protocolResult = canonicalizeProtocol(WTFMove(init.protocol), type);
+
+        if (protocolResult.hasException())
+            return protocolResult.releaseException();
+
+        result.protocol = protocolResult.releaseReturnValue();
+    }
+
+    if (!init.username.isNull())
+        result.username = canonicalizeUsername(WTFMove(init.username), type);
+
+    if (!init.password.isNull())
+        result.password = canonicalizePassword(WTFMove(init.password), type);
+
+    if (!init.hostname.isNull()) {
+        ExceptionOr<String> hostResult = canonicalizeHost(WTFMove(init.protocol), type);
+
+        if (hostResult.hasException())
+            return hostResult.releaseException();
+
+        result.hostname = hostResult.releaseReturnValue();
+    }
+
+    if (!init.port.isNull()) {
+        ExceptionOr<String> portResult = canonicalizePort(WTFMove(init.port), WTFMove(init.protocol), type);
+
+        if (portResult.hasException())
+            return portResult.releaseException();
+
+        result.port = portResult.releaseReturnValue();
+    }
+
+    if (!init.pathname.isNull()) {
+        result.pathname = init.pathname;
+
+        if (!baseURL.isNull() && baseURL.hasOpaquePath() && !isAbsolutePathname(result.pathname, type)) {
+            String baseURLPath = processBaseURLString(baseURL.path(), type);
+            size_t slashIndex = baseURLPath.reverseFind("/"_s);
+
+            if (slashIndex != WTF::notFound)
+                result.pathname = makeString(baseURLPath.substring(0, slashIndex + 1), result.pathname);
```

Comment:
use `StringView::left()`:
```
result.pathname = makeString(StringView { baseURLPath }.left(slashIndex + 1), result.pathname);
```

Reply:
`left()` is just nicer syntax for this use case. Using StringView() avoid allocating a String for no reason.